Package/gulpfile.js

```diff
@@ -0,0 +1,117 @@
+};
+
+const buildIphoneOS = async () => {
+  await exec('xcodebuild -sdk iphoneos -configuration Release -project ReactNativeBabylon.xcodeproj -scheme BabylonNative build CODE_SIGNING_ALLOWED=NO', {cwd: 'iOS/Build'});
+};
+
+const buildIphoneSimulator = async () => {
+  await exec('xcodebuild -sdk iphonesimulator -configuration Release -project ReactNativeBabylon.xcodeproj -scheme BabylonNative build CODE_SIGNING_ALLOWED=NO', {cwd: 'iOS/Build'});
+};
+
+const buildIOS = gulp.series(makeXCodeProj, buildIphoneOS, buildIphoneSimulator);
+
+const buildAndroid = async () => {
+  await exec('./gradlew babylonjs_react-native:assembleRelease', {cwd: '../Apps/Playground/android'});
+};
+
+const copyCommonFiles = async () => {
+  return  gulp.src('../Apps/Playground/node_modules/@babylonjs/react-native/package.json')
+    .pipe(gulp.src('../Apps/Playground/node_modules/@babylonjs/react-native/README.md'))
+    .pipe(gulp.src('../Apps/Playground/node_modules/@babylonjs/react-native/*.ts*'))
+    .pipe(gulp.src('react-native-babylon.podspec'))
+    .pipe(gulp.dest('Assembled'));
+};
+
+const copyIOSFiles = async () => {
+  return  gulp.src('../Apps/Playground/node_modules/@babylonjs/react-native/ios/*.h')
+    .pipe(gulp.src('../Apps/Playground/node_modules/@babylonjs/react-native/ios/*.mm'))
+    // This xcodeproj is garbage that we don't need in the package, but `pod install` ignores the package if it doesn't contain at least one xcodeproj. 🤷🏼‍♂️
+    .pipe(gulp.src('iOS/Build/ReactNativeBabylon.xcodeproj**/**/*'))
+    .pipe(gulp.dest('Assembled/ios'));
+};
+
+const createIOSUniversalLibs = async () => {
+  await exec('mkdir -p Assembled/ios/libs');
+  const libs = await readdirAsync('iOS/Build/Release-iphoneos');
+  await Promise.all(libs.map(lib => exec(`lipo -create iOS/Build/Release-iphoneos/${lib} iOS/Build/Release-iphonesimulator/${lib} -output Assembled/ios/libs/${lib}`)));
+};
+
+const copyAndroidFiles = async () => {
+  await new Promise(resolve => {
+          gulp.src('Android/build.gradle')
+    .pipe(gulp.src('../Apps/Playground/node_modules/@babylonjs/react-native/android/src**/main/AndroidManifest.xml'))
+    .pipe(gulp.src('../Apps/Playground/node_modules/@babylonjs/react-native/android/src**/main/java/**/*'))
+    .pipe(gulp.dest('Assembled/android'))
+    .on('end', resolve);
+  });
+
+  await new Promise(resolve => {
+          gulp.src('../Apps/Playground/node_modules/@babylonjs/react-native/android/build/intermediates/library_and_local_jars_jni/release/jni/**/*')
+    .pipe(gulp.dest('Assembled/android/src/main/jniLibs/'))
+    .on('end', resolve);
+  });
+};
+
+const createPackage = async () => {
+  await exec('npm pack', {cwd: 'Assembled'});
+};
+
+const copyFiles = gulp.parallel(copyCommonFiles, copyIOSFiles, copyAndroidFiles);
```

Comment:
I don't understand the difference between gulp.parallel and Promise.all. Why use one vs. the other?

Reply:
Gulp functions/tasks can return Promises, but they can also return a variety of other objects representing asynchronous work (such as those using the src/pipe/dest stuff). `gulp.parallel` understands all these, and also makes it so gulp understands the task hierarchy and can display reasonable command line output for the tasks that it is running.